src/lib.rs

```diff
@@ -27,7 +27,8 @@
 // External dependencies:
 //------------------------------------------------------------------------
 
-#[cfg(all(feature = "alloc", not(feature = "std")))]
+#[cfg(feature = "alloc")]
```

Comment:
Not for this PR, but shouldn't we have a thing where `std` is sufficient to use allocating functions? I've seen the `crate::prelude::Vec` pattern before for this

Reply:
`std` just re-exports what's in `alloc`, and the `std` crate feature also activates `alloc`.

This means that types defined in `alloc` are always available via `alloc`, regardless of whether or not the `std` feature is enabled.

The only historical reason to go through convolutions to access those types was when `alloc` wasn't stable.

Reply:
Ah ok that's good